### PR TITLE
CHE-8083 fix update custom stack flow

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@redhat.com>

### What does this PR do?
Fixed update custom stack flow.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/8083

#### Release Notes

N/A

### CODE REVIEW

**Review by ashumilova — APPROVED** [review]

**Review by akurinnoy — APPROVED** [review]

File: dashboard/src/app/stacks/stack-details/stack.controller.ts [hunk 1]
```diff
@@ -197,15 +197,9 @@ export class StackController {
...
-        this.stack = this.cheStack.getStackById(this.stackId);
-        this.prepareStackData();
-      } else {
-        this.$log.error(error);
-        this.loading = false;
-        this.invalidStack = error.statusText + error.status;
-      }
+      this.$log.error(error);
+      this.loading = false;
+      this.invalidStack = error.statusText + error.status;
```
Comment (benoitf) [inline dashboard/src/app/stacks/stack-details/stack.controller.ts]: do we need some extra separator between statusText and error status like a space or colon ?

File: dashboard/src/app/stacks/stack-details/stack.controller.ts [hunk 2]
```diff
@@ -303,11 +297,13 @@ export class StackController {
...
     }
     const stack = angular.fromJson(this.stackJson);
-    this.cheStack.updateStack(this.stack.id, stack).then((stack: any) => {
-      this.cheNotification.showInfo('Stack is successfully updated.');
-      this.isLoading = false;
-      this.stack = stack;
-      this.prepareStackData();
+    this.cheStack.updateStack(this.stack.id, stack).then(() => {
+      this.cheStack.fetchStacks().finally(() => {
+        this.cheNotification.showInfo('Stack is successfully updated.');
```
Comment (benoitf) [inline dashboard/src/app/stacks/stack-details/stack.controller.ts]: is it 'is' or we need verb to be in the "past"

**Review by benoitf — APPROVED** [review]

File: dashboard/src/app/stacks/stack-details/stack.controller.ts (hunk 1 above)
Comment (olexii4) [reply dashboard/src/app/stacks/stack-details/stack.controller.ts]: done

File: dashboard/src/app/stacks/stack-details/stack.controller.ts (hunk 2 above)
Comment (olexii4) [reply dashboard/src/app/stacks/stack-details/stack.controller.ts]: done

**Pull request merged**